### DISCUSSION TITLE
How to configure Minio with CDN?

### DISCUSSION THREAD

**Discussion by anthonytzvetkov** [minio/minio#16652] [Q&A] [answered]
Hello,

I am trying to configure MinIO with a CDN for awhile now, however i am coming short.

I have a test server on Ubuntu instance with Minio and Nginx installed.

I have set port forwarding on local router for port 9000 and 9090. My public IP is 82.146.20.102.

I have set the server's address in my CDN zone (13031-21.b.cdn12.com):

http://82.146.20.102:9090

I am starting MinIO with the following command:

sudo MINIO_ROOT_USER=$$$$ MINIO_ROOT_PASSWORD=$$$$ MINIO_PUBLIC_ADDRESS=82.146.20.102 minio server --address 192.168.0.103:9090 /data

My Nginx configuration file:

```
/etc/nginx/sites-enabled/minio                                                                         
server {
    listen 80;
    server_name 82.146.20.102;

    # To allow special characters in headers
    ignore_invalid_headers off;
    # Allow any size file to be uploaded.
    # Set to a value such as 1000m; to restrict file size to a specific value
    client_max_body_size 0;
    # To disable buffering
    proxy_buffering off;

    location / {
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header Host $http_host;

        proxy_connect_timeout 300;
        # Default is HTTP/1, keepalive is only enabled in HTTP/1.1
        proxy_http_version 1.1;
        proxy_set_header Connection "";
        chunked_transfer_encoding off;

        proxy_pass http://13031-21.b.cdn12.com;
    }

    location ~* \.(jpe?g|png|gif|ico|svg)$ {
        return 301 http://13031-21.b.cdn12.com$request_uri;
    }

    location /test/ {
        rewrite ^/test/(.*)$ http://13031-21.b.cdn12.com/test/$1 permanent;
    }
}
```

Now, if i access the CDN address, it successfully obtains the image from the origin and caches it:

http://13031-21.b.cdn12.com/test/AntonixBot_abusive_robot_and_girl_interrogation_tenticals_0788f6cd-7fef-46df-a84f-111086218d3c.png

But if i access the address directly, the request doesn't get proxied through the CDN and is provided by the origin:

http://82.146.20.102:9090/test/AntonixBot_abusive_robot_and_girl_interrogation_tenticals_0788f6cd-7fef-46df-a84f-111086218d3c.png

I am really new to this and trying to make this work for weeks now. Will appreciate some directions.

**Comment by AlexZIX** [comment, marked as answer]
You just should open only 80 port for access from internet and use 9090 only for administrative purposes inside your network. Also you should change 

proxy_pass http://13031-21.b.cdn12.com;

to 

proxy_pass http://192.168.0.103:9090;

or 

proxy_pass http://127.0.0.1:9090 in case of nginx and minio located at the same server.

and remove MINIO_PUBLIC_ADDRESS=82.146.20.102.

The way for accessing external users to your CDN looks like:

Client -> Internet -> Router -> Nginx -> Minio

Now you have two ways:

Client -> Internet -> Router -> Nginx -> Minio and Client -> Internet -> Router -> Minio in case of direct access to port 9090.

The main idea of reverse proxy is hide some severs behind this proxy and route requests via it.
